### ISSUE TITLE
Enhancement: allow amending user-agent with additional information

### ISSUE THREAD

**Issue by tswast** [hashicorp/terraform-provider-google#6832]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

The [Cloud Foundation Toolkit (CFT)](https://cloud.google.com/foundation-toolkit) provides many templates to help customers follow best practices when using this module. We would like to be able to track the usage of this toolkit by amending the user-agent header with the toolkit name and version.

Thus, instead of a user-agent of

```
terraform-provider-google/1.2.3
```

it would be helpful to have a user-agent of

```
cloud-foundation-toolkit/4.5.6 terraform-provider-google/1.2.3
```

Client libraries provide a similar user-agent option (which actually amends the user-agent as I suggest here). See the docs for the [Python user-agent config](https://googleapis.dev/python/google-api-core/latest/client_info.html).

### New or Affected Resource(s)

Affects all resources (transport-layer)

### Potential Terraform Configuration

Provide a `user_agent` option when instantiating a Google provider.

### References

* Relevant source code for current user-agent logic: https://github.com/terraform-providers/terraform-provider-google/blob/bb8099baa074a2156c9e2a689ee463a652f73670/google/config.go#L316

**Comment by rileykarson** [comment]
See https://github.com/hashicorp/terraform/pull/22583 which is the outcome of our longstanding feature request that will allow us to implement this, newly available in Terraform `0.13`. Once we implement the provider side it requires users to be using a version of TPG/TPGB with the new SDK (which we have not yet merged) as well as `0.13`.

**Comment by slevenick** [comment]
Available via https://github.com/GoogleCloudPlatform/magic-modules/pull/3995

@megan07 Do we have plans on where this will be documented?
